File: Java/Program/Penyewa.java
//Class Penyewa turunan dari Orang untuk simpan data tambahan atribut
public class Penyewa extends Orang {
    private String kodeSewa;
    private String tanggalMasuk;
    private String jatuhTempo;
    private String statusPembayaran;
    private Kosan kosan;
    private Kamar kamar;

    //Constructor kosong
    public Penyewa() {
        super();
        this.kodeSewa = "";
        this.tanggalMasuk = "";
        this.jatuhTempo = "";
        this.statusPembayaran = "";
        this.kosan = null;
        this.kamar = null;
    }

    //Constructor lengkap
    public Penyewa(long noKtp, String nama, String alamat, String noTelp, String kodeSewa, Kosan kosan, Kamar kamar, String tanggalMasuk, String jatuhTempo, String statusPembayaran) {
        super(noKtp, nama, alamat, noTelp);
        this.kodeSewa = kodeSewa;
        this.kosan = kosan;
        this.kamar = kamar;
        this.tanggalMasuk = tanggalMasuk;
        this.jatuhTempo = jatuhTempo;
        this.statusPembayaran = statusPembayaran;
    }

    //Getter & Setter khusus atribut Penyewa
    public String getKodeSewa(){
        return kodeSewa;
    }
    public void setKodeSewa(String kodeSewa){
        this.kodeSewa = kodeSewa;
    }

    public Kosan getKosan(){
        return kosan;
    }
    public void setKosan(Kosan kosan){
        this.kosan = kosan;
    }

    public Kamar getKamar(){
        return kamar;
    }
    public void setKamar(Kamar kamar){
        this.kamar = kamar;
    }

    public String getTanggalMasuk(){
        return tanggalMasuk;
    }
    public void setTanggalMasuk(String tanggalMasuk){
        this.tanggalMasuk = tanggalMasuk;
    }

    public String getJatuhTempo(){
        return jatuhTempo;
    }
    public void setJatuhTempo(String jatuhTempo){
        this.jatuhTempo = jatuhTempo;
    }

    public String getStatusPembayaran(){
        return statusPembayaran;
    }
    public void setStatusPembayaran(String statusPembayaran){
        this.statusPembayaran = statusPembayaran;
    }
}

File: Python/Program/Penyewa.py
from Orang import Orang
from Kamar import Kamar

#Class Penyewa turunan dari Orang untuk simpan data orang yang sewa
class Penyewa(Orang):
    def __init__(self, noKtp, nama, alamat, noTelp, kodeSewa, kosan, kamar: Kamar, tanggalMasuk, jatuhTempo, statusPembayaran):
        #Memanggil constructor Orang
        super().__init__(noKtp, nama, alamat, noTelp)
        self._kodeSewa = kodeSewa
        self._kosan = kosan
        self._kamar = kamar
        self._tanggalMasuk = tanggalMasuk
        self._jatuhTempo = jatuhTempo
        self._statusPembayaran = statusPembayaran

    #Getter & Setter untuk atribut khusus Penyewa
    def getKodeSewa(self):
        return self._kodeSewa
    def setKodeSewa(self, kodeSewa):
        self._kodeSewa = kodeSewa

    def getKosan(self):
        return self._kosan
    def setKosan(self, kosan):
        self._kosan = kosan

    def getKamar(self):
        return self._kamar
    def setKamar(self, kamar):
        self._kamar = kamar

    def getTanggalMasuk(self):
        return self._tanggalMasuk
    def setTanggalMasuk(self, tanggalMasuk):
        self._tanggalMasuk = tanggalMasuk

    def getJatuhTempo(self):
        return self._jatuhTempo
    def setJatuhTempo(self, jatuhTempo):
        self._jatuhTempo = jatuhTempo

    def getStatusPembayaran(self):
        return self._statusPembayaran
    def setStatusPembayaran(self, statusPembayaran):
        self._statusPembayaran = statusPembayaran